fecgov/fec-cms
# PR: 6179, 6180 national party account

## Summary

- Resolves #6179 
- Resolves #6180 

Building the front-end for the national party accounts' receipts and disbursements datatables

### Required reviewers

- UX
- Front-end for code
- Back-end and/or SMEs for data accuracy

## Impacted areas of the application

- A couple new feature flags
- A couple new URLs
- A new link from [/data/browse-data/?tab=raising](http://127.0.0.1:8000/data/browse-data/?tab=raising)
- A new link from [/data/browse-data/?tab=spending](http://127.0.0.1:8000/data/browse-data/?tab=spending)
- A couple new datatables
- I snuck in fixes for a few typos and cleaned up some page source

## Screenshots

| Receipts  | Disbursements |
| ------------- | ------------- |
| tab:![image](https://github.com/fecgov/fec-cms/assets/26720877/ae93752f-2619-4a97-a417-ca4a8d5219b9)  | tab:![image](https://github.com/fecgov/fec-cms/assets/26720877/db098631-f98c-4cc3-b7e6-b8bc84d6ab8a)  |
| datatable:![image](https://github.com/fecgov/fec-cms/assets/26720877/69b0c1f1-0eb7-40b8-b2f1-b3553248a336) | datatable:![image](https://github.com/fecgov/fec-cms/assets/26720877/7816b219-a73d-4f58-86cf-0d9a8a4e6883)  |

## Related PRs

Related PRs against other branches:

None for CMS

## Known issues

**UX and SMEs**: when the right-side details panel is open, it scoots down lower than the table. Should we lose a table column?

## How to test

- Pull the branch
- With env vars or directly in base.py,
   - make sure `FEC_API_URL` points to stage
   - add a valid API key
- `npm i`
- `npm run build`
- `./manage.py runserver`

Default, with feature flags on for localhost:

- [/data/browse-data/?tab=raising](http://127.0.0.1:8000/data/browse-data/?tab=raising) should offer an option for 'National party account receipts'
- [/data/browse-data/?tab=spending](http://127.0.0.1:8000/data/browse-data/?tab=spending) should offer an option for 'National party account disbursements'
- Both of those pages should link to their respective tables
- Check all filters
- Spot check a wide variety of lines' ➲ details panel

To test feature flag:
- For receipts
   1. base.py, line 113, comment out `FEATURES['nat_party_acct_receipts'] = True`, **save**
   1. [/data/browse-data/?tab=raising](http://127.0.0.1:8000/data/browse-data/?tab=raising) should no longer mention 'National party account receipts'
   1. The [national party account receipts datatable page](http://127.0.0.1:8000/data/national-party-account-receipts/) should throw a 404
   1. Restore the code in base.py, save
- For disbursements
   1. base.py, line 114, comment out `FEATURES['nat_party_acct_disbursements'] = True`, **save**
   1. [/data/browse-data/?tab=raising](http://127.0.0.1:8000/data/browse-data/?tab=spending) should no longer mention 'National party account disbursements'
   1. The [national party account disbursements datatable page](http://127.0.0.1:8000/data/national-party-account-disbursements/) should throw a 404
   1. Restore the code in base.py, save

## ⭐️ Next steps ⭐️

Linked tickets will need updates to stay in sync with the Webpack #2708 upgrade
